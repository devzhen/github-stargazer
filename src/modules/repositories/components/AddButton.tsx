import React from 'react';
import { Image } from 'react-native';
import styled from 'styled-components/native';
import { TapGestureHandler } from 'react-native-gesture-handler';

// @ts-ignore
import plusIcon from './assets/ic_plus.png';

interface AddButtonProps {
  onPress: any;
}

const AddButton = ({ onPress }: AddButtonProps) => (
  <TapGestureHandler onHandlerStateChange={onPress}>
    <Container>
      <Image
        source={plusIcon}
        style={{ height: 26, width: 26 }}
        resizeMode="contain"
      />
    </Container>
  </TapGestureHandler>
);

AddButton.defaultProps = {
  onPress: () => {},
};

const Container = styled.View`
  position: absolute;
  bottom: 25px;
  right: 10px;
  align-items: center;
  justify-content: center;
  width: 80px;
  height: 80px;
  border-radius: 40px;
  background-color: ${(props) => props.theme.colors.orange};
  box-shadow: 2px 2px 4px rgba(0, 0, 0, 0.4);
  z-index: 1000000;
`;

export default AddButton;
