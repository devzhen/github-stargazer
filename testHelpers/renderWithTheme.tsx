import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import 'jest-styled-components';

import theme from '@theme/index';

const renderWithTheme = (component: any) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export default renderWithTheme;
