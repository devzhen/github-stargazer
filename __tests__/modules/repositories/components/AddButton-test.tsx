import React from 'react';
import 'jest-styled-components';

import AddButton from '@modules/repositories/components/AddButton';

// @ts-ignore
import renderWithTheme from '@testHelpers/renderWithTheme';

describe('AddButton repository component', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<AddButton onPress={() => {}} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
