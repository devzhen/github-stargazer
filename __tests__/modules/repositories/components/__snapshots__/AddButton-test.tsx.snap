// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton repository component renders correctly 1`] = `
<View
  collapsable={false}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#eb9e38",
        "borderRadius": 40,
        "bottom": 25,
        "height": 80,
        "justifyContent": "center",
        "position": "absolute",
        "right": 10,
        "shadowColor": "rgba(0, 0, 0, 0.4)",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 4,
        "width": 80,
        "zIndex": 1000000,
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/modules/repositories/components/assets/ic_plus.png",
      }
    }
    style={
      Object {
        "height": 26,
        "width": 26,
      }
    }
  />
</View>
`;
